Regression in validate no longer enabling CRLF to be embedded within a Table_Character record

### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

Per [updates for](https://github.com/NASA-PDS/validate/pull/589) #519, added a regression for a failed test related to #325.

It appears that somewhere in the updates a regression was added where we are reading in `Table_Character` like `Table_Delimited`. So this special case (among probably others) are no longer handled correctly.

### 🕵️ Expected behavior

Validate successfully

### 📜 To Reproduce

Previously when running `mvn test`:

```
[ERROR] Execute validate command for tests below. #87  Time elapsed: 0.755 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 0 error messages expected. {"title":"PDS Validation Tool Report","configuration":{"version":"3.2.0-SNAPSHOT","date":"2023-01-30T22:02:41Z"},"parameters":{"targets":"[file:/Users/jpadams/proj/pds/pdsen/workspace/validate/src/test/resources/github325/crs009x.xml]","ruleType":"pds4.label","severityLevel":"WARNING","recurseDirectories":"true","fileFiltersUsed":"[*.xml, *.XML]","dataContentValidation":"on","productLevelValidation":"on","maxErrors":"100000","registeredContextsFile":"/Users/jpadams/proj/pds/pdsen/workspace/validate/src/main/resources/util/registered_context_products.json"},"productLevelValidationResults":[{"status":"FAIL","label":"file:/Users/jpadams/proj/pds/pdsen/workspace/validate/src/test/resources/github325/crs009x.xml","messages":[],"fragments":[],"dataContents":[{"dataFile":"file:/Users/jpadams/proj/pds/pdsen/workspace/validate/src/test/resources/github325/crs009x.tab","messages":[{"severity":"ERROR","type":"error.table.record_length_mismatch","table":1,"record":1,"message":"Delimiter is not at the end of the record. Record read using delimiter is 82 bytes long while record is defined to be 246 bytes."},{"severity":"ERROR","type":"error.table.record_length_mismatch","table":1,"record":2,"message":"Delimiter is not at the end of the record. Record read using delimiter is 82 bytes long while record is defined to be 246 bytes."}]}]}],"summary":{"totalProducts":1,"totalErrors":2,"totalWarnings":0,"messageTypes":[{"messageType":"error.table.record_length_mismatch","total":2}]}} ==> expected: <2> but was: <0>
```

Now, executing validate against that file in the repo, instead of the test executing successfully, we get 2 errors:
```
$ validate -t src/test/resources/github325/crs009x.xml
```

### 🖥 Environment Info

- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...


### 📚 Version of Software Used

_No response_

### 🩺 Test Data / Additional context

See repo path in execution above

### 🦄 Related requirements

_No response_

### ⚙️ Engineering Details

_No response_

--- a/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
@@ -150,7 +150,8 @@ public class TableValidator implements DataObjectValidator {
       if (this.tableAdapter instanceof TableBinaryAdapter) {
         this.validateTableBinaryContent(fieldValueValidator, record, spotCheckData,
             keepQuotationsFlag);
-      } else if (!this.isTableLineOriented() && !this.getCheckInbetweenFields()) {
+      } else if (this.tableAdapter instanceof TableDelimitedAdapter &&
+    		  !this.isTableLineOriented() && !this.getCheckInbetweenFields()) {
         // Determine if we should proceed with calling validateTableDelimited()
         // function.
         // Note that the function validateTableDelimited() can only be applied if the
@@ -444,7 +445,7 @@ public class TableValidator implements DataObjectValidator {
     // reduced the function size.
     TableCharacterUtil tableCharacterUtil = null;
     boolean manuallyParseRecord = false;
-    String line = this.currentTableReader.readNextLine();
+    String line = this.currentTableReader.readNextFixedLine();
     long lineNumber = 0;
     int dataObjectIndex = this.tableObject.getDataObjectLocation().getDataObject();
 
@@ -609,7 +610,7 @@ public class TableValidator implements DataObjectValidator {
         break;
       }
 
-      line = this.currentTableReader.readNextLine();
+      line = this.currentTableReader.readNextFixedLine();
       if (line != null) {
         LOG.debug("recordNumber: {}, line.length: {}, record: [{}]", lineNumber, line.length(),
             line);

--- a/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
@@ -19,6 +19,7 @@ import java.net.URL;
 import java.nio.BufferUnderflowException;
 import java.util.ArrayList;
 import java.util.Map;
+import org.checkerframework.common.returnsreceiver.qual.This;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import com.opencsv.exceptions.CsvValidationException;
@@ -445,7 +446,7 @@ public class TableValidator implements DataObjectValidator {
     // reduced the function size.
     TableCharacterUtil tableCharacterUtil = null;
     boolean manuallyParseRecord = false;
-    String line = this.currentTableReader.readNextFixedLine();
+    String line = this.tableAdapter instanceof TableDelimitedAdapter ? this.currentTableReader.readNextLine() : this.currentTableReader.readNextFixedLine();
     long lineNumber = 0;
     int dataObjectIndex = this.tableObject.getDataObjectLocation().getDataObject();
 
@@ -610,7 +611,7 @@ public class TableValidator implements DataObjectValidator {
         break;
       }
 
-      line = this.currentTableReader.readNextFixedLine();
+      line = this.tableAdapter instanceof TableDelimitedAdapter ? this.currentTableReader.readNextLine() : this.currentTableReader.readNextFixedLine();
       if (line != null) {
         LOG.debug("recordNumber: {}, line.length: {}, record: [{}]", lineNumber, line.length(),
             line);

--- a/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
@@ -449,7 +449,7 @@ public class TableValidator implements DataObjectValidator {
     // reduced the function size.
     TableCharacterUtil tableCharacterUtil = null;
     boolean manuallyParseRecord = false;
-    String line = this.tableAdapter instanceof TableDelimitedAdapter ? this.currentTableReader.readNextLine() : this.currentTableReader.readNextFixedLine();
+    String line;
     long lineNumber = 0;
     int dataObjectIndex = this.tableObject.getDataObjectLocation().getDataObject();
 
@@ -457,11 +457,9 @@ public class TableValidator implements DataObjectValidator {
     // (ending with .tab) but with the table type.
     // If the type of the table is not TableDelimited, the checking of same line
     // length should be done.
-    boolean tableIsFixedLength = true;
+    boolean tableIsFixedLength = !(this.tableAdapter instanceof TableDelimitedAdapter);
 
-    if (this.tableAdapter instanceof TableDelimitedAdapter) {
-      tableIsFixedLength = false;
-    } else {
+    if (tableIsFixedLength) {
       tableCharacterUtil = new TableCharacterUtil(this.context.getTarget(), this.listener);
       tableCharacterUtil.parseFieldsInfo();
     }
@@ -471,6 +469,8 @@ public class TableValidator implements DataObjectValidator {
     // Add 2 arrays to keep track of each line number and its length.
     ArrayList<Integer> lineLengthsArray = new ArrayList<>(0);
     ArrayList<Long> lineNumbersArray = new ArrayList<>(0);
+
+    line = tableIsFixedLength ? this.currentTableReader.readNextFixedLine() : this.currentTableReader.readNextLine();
 
     if (line != null) {
       LOG.debug("validateTableCharacter:POSITION_1:lineNumber,line {},[{}],{}", lineNumber, line,
@@ -614,7 +614,7 @@ public class TableValidator implements DataObjectValidator {
         break;
       }
 
-      line = this.tableAdapter instanceof TableDelimitedAdapter ? this.currentTableReader.readNextLine() : this.currentTableReader.readNextFixedLine();
+      line = tableIsFixedLength ? this.currentTableReader.readNextFixedLine() : this.currentTableReader.readNextLine();
       if (line != null) {
         LOG.debug("recordNumber: {}, line.length: {}, record: [{}]", lineNumber, line.length(),
             line);

--- a/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java
@@ -19,7 +19,6 @@ import java.net.URL;
 import java.nio.BufferUnderflowException;
 import java.util.ArrayList;
 import java.util.Map;
-import org.checkerframework.common.returnsreceiver.qual.This;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import com.opencsv.exceptions.CsvValidationException;

--- a/src/test/resources/features/validate.feature
+++ b/src/test/resources/features/validate.feature
@@ -267,12 +267,12 @@ Scenario Outline: Execute validate command for tests below.
 
 # https://github.com/nasa-pds/validate/issues/379 FileService:printStackTraceToFile:ERROR when validating a product with overlapping fields
 
- |"NASA-PDS/validate#379 INVALID" | "github379" | 1 | "1 errors message expected" | "totalErrors" | "src/test/resources" | "target/test" | "-R pds4.label --skip-context-validation --skip-context-reference-check -r {reportDir}/report_github379_label_invalid.json   -s json -t {resourceDir}/github379/mix_cal_hk_fpac_report_20181204.xml" | "report_github379_label_invalid.json" |
+ |"NASA-PDS/validate#379 INVALID" | "github379" | 3 | "3 errors message expected" | "totalErrors" | "src/test/resources" | "target/test" | "-R pds4.label --skip-context-validation --skip-context-reference-check -r {reportDir}/report_github379_label_invalid.json   -s json -t {resourceDir}/github379/mix_cal_hk_fpac_report_20181204.xml" | "report_github379_label_invalid.json" |
 
 # https://github.com/nasa-pds/validate/issues/380 stack trace being created during successful validate execution
 
- |"NASA-PDS/validate#380 INVALID_1" | "github380" | 1 | "1 errors message expected: Reuse github379 resources" | "totalErrors" | "src/test/resources" | "target/test" | "-R pds4.label --skip-context-validation --skip-context-reference-check -r {reportDir}/report_github380_label_invalid_without_stack.json   -s json -t {resourceDir}/github379/mix_cal_hk_fpac_report_20181204.xml" | "report_github380_label_invalid_without_stack.json" |
- |"NASA-PDS/validate#380 INVALID_2" | "github380" | 1 | "1 errors message expected: Reuse github379 resources" | "totalErrors" | "src/test/resources" | "target/test" | "-R pds4.label --debug-mode --skip-context-validation --skip-context-reference-check -r {reportDir}/report_github380_label_invalid_without_stack.json   -s json -t {resourceDir}/github379/mix_cal_hk_fpac_report_20181204.xml" | "report_github380_label_invalid_without_stack.json" |
+ |"NASA-PDS/validate#380 INVALID_1" | "github380" | 3 | "3 errors message expected: Reuse github379 resources" | "totalErrors" | "src/test/resources" | "target/test" | "-R pds4.label --skip-context-validation --skip-context-reference-check -r {reportDir}/report_github380_label_invalid_without_stack.json   -s json -t {resourceDir}/github379/mix_cal_hk_fpac_report_20181204.xml" | "report_github380_label_invalid_without_stack.json" |
+ |"NASA-PDS/validate#380 INVALID_2" | "github380" | 3 | "3 errors message expected: Reuse github379 resources" | "totalErrors" | "src/test/resources" | "target/test" | "-R pds4.label --debug-mode --skip-context-validation --skip-context-reference-check -r {reportDir}/report_github380_label_invalid_without_stack.json   -s json -t {resourceDir}/github379/mix_cal_hk_fpac_report_20181204.xml" | "report_github380_label_invalid_without_stack.json" |
  
 # https://github.com/nasa-pds/validate/issues/375 validate halts if label has name "collection" embedded 
 

--- a/src/test/resources/features/validate.feature
+++ b/src/test/resources/features/validate.feature
@@ -199,7 +199,7 @@ Scenario Outline: Execute validate command for tests below.
 
 # https://github.com/NASA-PDS/validate/issues/325 Validate Incorrectly Throws Error When Embedded Field_Character Contains <CR><LF>
 # temp turn off and fix again in new issue because character table seems to be getting processed as a delimited table
-# |"NASA-PDS/validate#325 VALID" | "github325" | 0 | "0 error messages expected." | "totalErrors" | "src/test/resources" | "target/test" | "-r {reportDir}/report_github325_label_valid.json -s json -R pds4.label -t {resourceDir}/github325/crs009x.xml" | "report_github325_label_valid.json" |
+ |"NASA-PDS/validate#325 VALID" | "github325" | 0 | "0 error messages expected." | "totalErrors" | "src/test/resources" | "target/test" | "-r {reportDir}/report_github325_label_valid.json -s json -R pds4.label -t {resourceDir}/github325/crs009x.xml" | "report_github325_label_valid.json" |
 
 # https://github.com/NASA-PDS/validate/issues/335 validate gives a NullPointerException during validation of a directory containing Table_Character products
 

